opsdroid/connector/twitch/__init__.py

```diff
@@ -0,0 +1,576 @@
+        
+        async with aiohttp.ClientSession() as session:
+            
+            payload = {
+                "hub.callback": f"{self.forward_url}/connector/{self.name}", 
+                "hub.mode": mode,
+                "hub.topic": topic,
+                "hub.lease_seconds": 60 * 60 * 24,
+            }
+            
+            response = await session.post(
+                TWITCH_WEBHOOK_ENDPOINT, 
+                headers=headers, 
+                json=payload
+            )
+
+    async def handle_challenge(self, request):
+        """Challenge handler for get request made by Twitch.
+        
+        Upon subscription to a Twitch webhook, Twitch will do a get request to the 
+        `callback` url provided to check if the url exists. Twitch will do a get request
+        with a challenge and expects the `callback` url to return that challenge in plain-text
+        back to Twitch.
+        
+        This is what we are doing here, we are getting `hub.challenge` from the request and return
+        it in plain-text, if we can't find that challenge we will return a status code 500.
+        
+        Args:
+            request (aiohttp.web.Request): Request made to the get route created for webhook subscription.
+        
+        Returns:
+           aiohttp.web.Response: if request contains `hub.challenge` we return it, otherwise return status 500.
+        
+        """
+        
+        challenge = request.rel_url.query.get('hub.challenge')
+
+        if challenge:
+            return aiohttp.web.Response(text=challenge)
+        return aiohttp.web.Response(status=500)
+
+    async def twitch_webhook_handler(self, request):
+        """Handle event from Twitch webhooks.
+        
+        This method will handle events when they are pushed to the webhook post route. Each webhook will
+        send a different kind of payload so we can handle each event and trigger the right opsdroid event
+        for the received payload.
+        
+        For follow events the payload will contain `from_id`(broadcaster id), `from_username`(broadcaster username)
+        `to_id`(follower id), `to_name` (follower name) and `followed_at` (timestamp).
+        
+        For stream changes a lot more things are returned but we only really care about `type`(if live/offline), 
+        `title`(stream title).
+        
+        For subscriptions events we will want to know `event_type`, `timestamp`, `event_data.plan_name`, `event_data.is_gift`,
+        `event_data.tier`, `event_data.username` and `event_data.gifter_name`.
+        
+        Args:
+            request (aiohttp.web.Request): Request made to the post route created for webhook subscription.
+
+        Return:
+            aiohttp.web.Response: Send a `received` message and status 200 - Twitch will keep sending the event if 
+            it doesn't get the 200 status code. 
+
+        """
+        payload = await request.json()
+        data = payload.get('data')[0]
```

Comment:
If the list is only ever one item it is more Pythonic to unpack at assingment like this

```suggestion
        [data] = payload.get('data')
```